app/controllers/subheadings_controller.rb

```diff
@@ -1,6 +1,8 @@
 class SubheadingsController < GoodsNomenclaturesController
   before_action :set_goods_nomenclature_code, only: %i[show]
 
+  rescue_from Faraday::ResourceNotFound, with: :show_validity_periods
+
```

Comment:
to be more explicit, I think it's better to put this close to the call that could trigger that exception

Reply:
Discussed this one in person - basically we're copying an existing pattern